assets/js/modules/analytics/datastore/properties.test.js

```diff
@@ -123,6 +123,67 @@ describe( 'modules/analytics properties', () => {
 				expect( properties ).toEqual( undefined );
 			} );
 		} );
+		describe( 'selectProperty', () => {
+			it( 'requires a valid propertyID', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).selectProperty();
+				} ).toThrow( 'A valid propertyID selection is required.' );
+			} );
+
+			it( 'returns if the accountID is not set', () => {
+				fetchMock.get(
+					/^\/google-site-kit\/v1\/modules\/analytics\/data\/properties-profiles/,
+					{ body: fixtures.propertiesProfiles, status: 200 }
+				);
+				const propertyID = fixtures.propertiesProfiles.properties[ 0 ].id;
+				expect( registry.select( STORE_NAME ).getPropertyID() ).toBeUndefined();
+				registry.dispatch( STORE_NAME ).selectProperty( propertyID );
+				expect( registry.select( STORE_NAME ).getPropertyID() ).toBeUndefined();
+			} );
+
+			it( 'selects the property', async () => {
+				fetchMock.get(
+					/^\/google-site-kit\/v1\/modules\/analytics\/data\/properties-profiles/,
+					{ body: fixtures.propertiesProfiles, status: 200 }
+				);
+				const accountID = fixtures.propertiesProfiles.properties[ 0 ].accountId;
+				const propertyID = fixtures.propertiesProfiles.properties[ 0 ].id;
+				registry.dispatch( STORE_NAME ).setAccountID( accountID );
+				registry.dispatch( STORE_NAME ).selectProperty( propertyID );
+
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getProperties( accountID ) !== undefined
+					),
+				);
```

Comment:
We only need to wait for resolvers for selectors. For async actions, we can simply `await` the dispatch 🙂 

```suggestion
				await registry.dispatch( STORE_NAME ).selectProperty( propertyID );

```